firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/CustomKeysAndValues.java

```diff
@@ -0,0 +1,70 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.crashlytics.internal.common;
+
+import java.util.HashMap;
+import java.util.Map;
+import androidx.annotation.NonNull;
+
+/**
+ * Helper class which handles the storage and conversion to strings of key/value
+ * pairs with heterogenous value types.
+ */
+public class CustomKeysAndValues {
+
+  public final Map<String, String> keysAndValues;
+
+  static class Builder {
+
+    // Holds the converted pairs of custom keys and values.
+    Map<String, String> keysAndValues = new HashMap<String, String>();
```

Comment:
Won't this allow someone to reset it's value from another class?
(Eg. by calling `CustomKeysAndValues.Builder.keysAndValues = new HashMap<String, String>()`)